Code/Framework/AzToolsFramework/AzToolsFramework/FocusMode/FocusModeSystemComponent.cpp

```diff
@@ -126,9 +128,30 @@ namespace AzToolsFramework
 
     void FocusModeSystemComponent::OnEntityInfoUpdatedAddChildEnd(AZ::EntityId parentId, AZ::EntityId childId)
     {
-        // If the parent's entityId is in the list, add the child.
-        if (auto iter = AZStd::find(m_focusedEntityIdList.begin(), m_focusedEntityIdList.end(), parentId);
-            iter != m_focusedEntityIdList.end())
+        // If the parent's entityId is in the list and the child isn't, add the child to the list.
+        bool isParentInList = false;
+        bool isChildInList = false;
+
+        for (auto iter = m_focusedEntityIdList.begin(); iter != m_focusedEntityIdList.end(); ++iter)
```

Comment:
question: Any reason why we're not just using a range based for loop here? 😛 This is so 1998 :wink:

Reply:
This could still just be...

```
for (const auto focusedEntityId : m_focusedEntityIdList)
{
    if (focusedEntityId == parentId)
    {
        isParentInList = true;
    }

    if (focusedEntityId == childId)
    {
        isChildInList = true;
    }

    // Early out
    if (isChildInList)
    {
        break;
    }
}
```
Or am I missing something silly sorry?

Reply:
You are correct, I think I missed something silly when I refactored the function multiple times :D
I will refactor it next time I get back to that function :)